### DISCUSSION TITLE
Is there a way to expose onConnectionStateChange in callbacks for a connected peripheral?

### DISCUSSION THREAD

**Discussion by eraguzin** [JuulLabs/kable#566] [Q&A]
In native Android BLE, when using the [connectGatt()](https://developer.android.com/reference/android/bluetooth/BluetoothDevice) function, you can pass in a [BluetoothGattCallback object](https://developer.android.com/reference/android/bluetooth/BluetoothGattCallback) which allows you to receive callbacks not only for service and characteristic operations, but to see when the device connects or disconnects.

I see that kable has an internal [onConnectionStateChange callback](https://github.com/JuulLabs/kable/blob/89b9b031c977c006c778b164e6e0f054f1638d56/core/src/androidMain/kotlin/gatt/Callback.kt#L100), but I don't think it's exposed in the API. I see the logger run in the debug when I connect, but not when I disconnect. 

Basically, if the other BLE device I'm connected to disconnects (Whether because I chose to or it disconnects abruptly on the other end), I'd like to get a callback so I can update the GUI, etc... with that information. Is there currently a way to do that with Kable?

**Comment by eraguzin** [comment]
Never mind, I see the "State" section in the docs, sorry.
